app/javascript/app/components/simple-menu/simple-menu-component.jsx

```diff
@@ -13,46 +14,91 @@ class SimpleMenu extends PureComponent {
   constructor() {
     super();
     this.state = {
-      open: false
+      open: false,
+      actionSuccessful: false
     };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(action) {
+    this.setState({ actionSuccesful: action() });
+  }
+  renderInsideLink(option, withAction = false) {
+    const { actionSuccesful } = this.state;
+    return (
+      <div className={styles.documentLink} key={option.label}>
+        {option.icon &&
+          (withAction && actionSuccesful ? (
+            <Icon icon={checkIcon} className={styles.icon} />
```

Comment:
This is unnecessary. You can do the ternary inside the icon prop for more clarity and less code.